Python/ceval.c

```diff
@@ -4603,16 +4635,44 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, InterpreterFrame *frame, int thr
 
         TARGET(CALL_FUNCTION): {
             PREDICTED(CALL_FUNCTION);
-            PyObject **sp, *res;
-            sp = stack_pointer;
-            res = call_function(tstate, &sp, oparg, NULL, cframe.use_tracing);
-            stack_pointer = sp;
-            PUSH(res);
-            if (res == NULL) {
+            PyObject *res;
+
+            // Check if the call can be inlined or not
+            PyObject *function = *(stack_pointer - oparg - 1);
+            int inline_call = 0;
+            if (Py_TYPE(function) == &PyFunction_Type) {
+                PyCodeObject *code = (PyCodeObject*)((PyFunctionObject*)(function))->func_code;
```

Comment:
Apologies in advance if the macro isn't available in ceval:
```suggestion
                PyCodeObject *code = (PyCodeObject*)PyFunction_GET_CODE(function);
```